api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -369,23 +380,23 @@ public T get() {
 
   static int decimalMaxPrecision(int numBytes) {
     Preconditions.checkArgument(numBytes >= 0 && numBytes < 24,
-        "Unsupported decimal length: " + numBytes);
+        "Unsupported decimal length: %d", numBytes);
```

Comment:
Preconditions format should use only `%s`. See https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Preconditions.java#L100-L105